core/cc/id.h

```diff
@@ -51,7 +51,12 @@ namespace std {
 
 template <> struct hash<core::Id> {
     inline size_t operator()(const core::Id& id) const {
-        return *reinterpret_cast<const size_t*>(id.data);
+        union {
```

Comment:
IIRC this breaks strict aliasing.
The only way to do this kind of cast "legally" is
```
static_assert(sizeof(core::Id*) == sizeof(size_t), "Incompatible types"); // Sanity check
size_t hash;
memcpy(&hash, &id, sizeof(size_t));
return hash;

All of the compilers SHOULD optimize the memcpy away, and this becomes a no-op anyway.

Reply:
Done. My memory is set on telling me that the `union` is the correct solution, regardless on how many times you correct me. :)

Reply:
Looks good!